docs/rhoas/iac_tools_rhoas/README.adoc

```diff
@@ -0,0 +1,703 @@
+
+[NOTE]
+The offline token is a refresh token with no expiry and can be used by non-interactive processes to provide an access token for {product-long-rhoas} to authenticate the user. The token is an OpenShift offline token and can be found at https://cloud.redhat.com/openshift/token. As the offline token is a sensitive value that varies between environments it is best specified through as `_OFFLINE_TOKEN_` environment variable when running `terraform apply` in a terminal.
+
+.Procedure
+. Open your browser of choice and navigate to the RHOAS Terraform provider.
+. Click *Use Provider*.
+. Copy the code provided.
++
+
+.Example rhoas provider configuration
+[source,shell]
+----
+$ terraform {
+  required_providers {
+    rhoas = {
+      source = "redhat-developer/rhoas"
+      version = "0.3.0"
+    }
+  }
+}
+
+provider "rhoas" {
+  #configuration options
+}
+----
+
+. Open your IDE of choice and paste the code into it. You can specify configuration options in the provider settings. For example, you can enter the offline token here using the `offline_token` field. However as noted previously, it is recommended to enter the offline token as an environment variable.
++
+
+. Save the file as a .tf file.
+. Open a terminal and enter the following command in the Terraform directory. This command initializes the working directory containing Terraform configuration files and installs any required plug-ins.
++
+[source,shell]
+----
+$ terraform init
+----
+A success message displays that `Terraform has been successfully initialized!`
+
+[id="con-using-terraform_{context}"]
+=== Using the RHOAS Terraform provider
+
+Resources are the most important element in the Terraform language. Each resource block describes one or more infrastructure objects. Terraform has error messages that help with minor typing errors by suggesting a resource close to what is typed. The next steps show what resources (either for single tasks or a combination of tasks) you can add to your Terraform configuration file to provision your Kafka instance.
+
+[id"proc-creating-kafka-instance-terraform_{context}"]
+[discrete]
+=== Creating a Kafka instance
+
+[role="_abstract"]
+To manage a Kafka instance using Terraform, use the `rhoas_kafka` resource.
+
+.Prerequisites
+
+* You have an offline token that is accessible to Terraform.
+* You have a Red Hat account.
+
+.Procedure
+
+. Open your IDE of choice and enter the example rhoas_kafka resource. This example uses the “instance”  identifier and creates a Kafka instance called `my-instance`. The only required fields are the `name`, `billing_model` and `plan` fields. The following values are set by default when you run `terraform apply`. All other information for the instance is provided by the Kafka APIs.
++
+.. `cloud provider`: A list of available cloud providers can be obtained using `data.rhoas_cloud_providers`.
+.. `region`: A list of available regions can be obtained using
+`data.rhoas_cloud_providers_regions`
```

Comment:
I was a bit confused by this:

- When we say "In your IDE, enter the example `rhoas_kafka` resource" what do we mean by _enter_? Do we mean open a  new file and copy and paste the configuration shown below?
- What do we mean by obtaining values using `data.rhoas_cloud_providers`? Where/how do we do this? - 
- Where do we apply the `billing_model`, `plan`, `cloud provider`, `region` parameters? I don't see any of these in the example shown below.

Overall, I couldn't quite tell if I'm looking at a complete, usable Terraform resource, or you're instructing me on how to configure one? If it's the latter, then I think we need to be much granular (with steps) about how to gather and add all the necessary parameters to the resource.